WcaOnRails/spec/factories/persons.rb

```diff
@@ -3,9 +3,9 @@
   factory :person do
     wca_id do
       mid = ('A'..'Z').to_a.sample(4).join
-      "2016#{mid}01".tap do |id|
-        id.next! while Person.exists?(wca_id: id)
-      end
+      id = "2016#{mid}01"
+      id = id.next while Person.exists?(wca_id: id)
+      id
```

Comment:
I had to leave the explicit return `id` at the end because line 7 (with the inline while loop) returns `nil`:

```
irb(main):006:0> i = 1
=> 1
irb(main):007:0> i = 3 while i == 1
=> nil
irb(main):008:0> i
=> 3
```

@larspetrus, I fear your solution becomes a problem once num equals 99, whereas `.next` will do a cool thing:

```
irb(main):009:0> "2016FLEI99".next
=> "2016FLEJ00"
```